app/controllers/course/discussion/posts_controller.rb

```diff
@@ -0,0 +1,106 @@
+class Course::Discussion::PostsController < Course::ComponentController
+  before_action :set_topic
```

Comment:
But that won't initiate an instance variable `@conditional` which might be used in views.